### PR TITLE
add text for NamePopover of Guild Folders if blank

### PR DESCRIPTION
**Pull request by jy1263**
i'm not entirely sure if this is the best way of doing it, but added it to emulate discord's behavior with blank folder names.

### CODE REVIEW

File: internal/gtkcord/sidebar/guilds/folder.go [hunk 1]
```diff
@@ -158,6 +157,22 @@ func (f *Folder) Set(folder *gateway.GuildFolder) {
 	for _, g := range f.Guilds {
 		g.Invalidate()
 	}
+
+	// After guilds are loaded, read their labels and set the folder name if unset.
+	if len(folder.Name) == 0 {
```
Comment (diamondburned) [inline internal/gtkcord/sidebar/guilds/folder.go:162]: You should do `folder.Name == ""` here, as it's more consistent with the rest of the code.

File: internal/gtkcord/sidebar/guilds/folder.go [hunk 2]
```diff
@@ -158,6 +157,22 @@ func (f *Folder) Set(folder *gateway.GuildFolder) {
...
+		for i, g := range f.Guilds {
```
Comment (diamondburned) [inline internal/gtkcord/sidebar/guilds/folder.go:163]: It'd be better if you do

```go
var name string
for ... {
    name += "..."
}
f.Name.SetName(name)
```

Modifying `*gateway.GuildFolder` may not be the best idea.

Comment (jy1263) [reply internal/gtkcord/sidebar/guilds/folder.go:163]: okay will do, sorry am a bit new to go, didn't know funcs were pass by reference

**Comment by jy1263** [comment]
adapted code for both reviews

**Review by diamondburned — APPROVED** [review]

**Pull request merged**